### DISCUSSION TITLE
sqlalchemy.exc.OperationalError: (pyodbc.OperationalError) ('08001', '[08001] [Microsoft][ODBC Driver 17 for SQL Server]命名管道提供程序: 无法打开与 SQL Server 的连接 [53]

### DISCUSSION THREAD

**Discussion by wangzhenhenan** [sqlalchemy/sqlalchemy#9780] [Usage Questions]
用pyodbc可以连接，在sqlalchemy连接不上，驱动也没错 ，感觉官方对mssql支持很差

**Comment by CaselIT** [comment]
Hi, sorry but we don't understand Chinese.

Trying to convert the message yields:

> sqlalchemy.exc.OperationalError: (pyodbc.OperationalError) (‘08001’, '[08001] [Microsoft][ODBC Driver 17 for SQL Server]Named Pipes Provider: Could not open a connection to SQL Server [53]

> I can connect with pyodbc, but I can’t connect with sqlalchemy. The driver is also correct. I feel that the official support for mssql is very poor.

No information is given, so I don't know what the issue is.
If you can connect with pyodb make sqlalchemy use that connection: https://docs.sqlalchemy.org/en/20/core/engines.html#fully-replacing-the-dbapi-connect-function
